ganga/GangaGUI/gui/routes.py

```diff
@@ -64,13 +83,190 @@ def decorated(*args, **kwargs):
+
+    # Get job from jobs repository
+    try:
+        j = jobs[job_id]
+        response_data = {attribute: str(getattr(j, attribute))}
+    except Exception as err:
+        return jsonify({"success": False, "message": str(err)}), 400
+
+    return jsonify(response_data)
+
+
+# Create job using existing template API - POST Method
+@app.route("/job/create", methods=["POST"])
+@token_required
+def job_create_endpoint(current_user):
+    """
+    API to create a job using the existing template.
+
+    IMPORTANT: template_id NEEDS to be provided in the request body. job_name can also be provided in the request body.
+    """
+
+    # Imports
+    from GangaCore.GPI import templates, Job
+
+    # Store request data
+    template_id: int = request.form.get("template_id")
+    job_name: str = request.form.get("job_name")
+
+    # template_id existence check
+    if template_id is None:
+        return jsonify({"success": False,
+                        "message": "Template ID not provided in the request data"}), 400
 
+    # Create job using template
+    try:
+        j = Job(templates[int(template_id)])
+        if job_name is not None:
+            j.name = job_name
+    except Exception as err:
+        return jsonify({"success": False, "message": str(err)}), 400
+
+    return jsonify({"success": True,
+                    "message": "Job with ID {} created successfully using the template ID {}".format(j.id,
+                                                                                                     template_id)})
+
+
+# Perform certain action on the Job - PUT Method
+@app.route("/job/<int:job_id>/<action>", methods=["PUT"])
+# @token_required
+def job_action_endpoint(job_id: int, action: str):
+    """
+    Given the job_id and action in the endpoint, perform the action on the job.
+
+    Supported actions: do_auto_resubmit, name, parallel_submit, copy, kill, force_status, resubmit, runPostProcessors, submit
+
+    Important Info:
+    1) do_auto_resubmit: User needs to provide 'do_auto_resubmit' field and it's value in the request body.
+    2) name: User needs to provide 'name' field and it's value in the request body.
+    3) parallel_submit: User needs to provide 'parallel_submit' field and it's value in the request body.
+    4) force_status: User needs to provide 'force_status' field and it's value in the request body.
+
+    :param job_id: int
+    :param action: str
+    :param attribute: str
+    """
+
+    # Imports
+    from GangaCore.GPI import jobs
+    from GangaCore.GPIDev.Lib.Job import Job
+
+    # Action support check
+    if action not in Job._exportmethods:
+        return jsonify({"success": False,
+                        "message": "Job action {} is not currently supported or does not exist".format(action)}), 400
+
+    request_data = request.form.to_dict()
+
+    try:
+        j = jobs(job_id)
+        if action in request_data.keys():
+            getattr(j, action)(request_data[action])
```

Comment:
How does this work if there is an optional argument? Like the `resubmit` or `peek` methods?